greenriver/hmis-frontend
# Add CI step for storybook test runner

## Description

GH issue: https://github.com/open-path/Green-River/issues/6145

Summary of changes:
- Add a CI step for the Storybook test runner
- Add a simple test that checks, for each story, whether it has any a11y violations.
  - Example build that shows the passing tests: https://github.com/greenriver/hmis-frontend/actions/runs/13317749398/job/37195788692?pr=935
- Disable certain violations, particularly contrast check, which we should roll back after our theme updates are introduced.*
  - To see how it looks when tests fail, locally, you can comment out [these lines](https://github.com/greenriver/hmis-frontend/pull/935/files#diff-0bf5908655f9e530c64c03d6084091a38f86e59e75f9266134bdf27513e2e7e8R16-R20), then run `yarn test-storybook`
- Add a note to the README about how to run a11y tests locally
- Fix a few minor issues that were causing a11y tests to fail

*Note that these violations are _not_ disabled on the Storybook Stories themselves, only in testing. So you can still see them when you go to the story:
![Screenshot 2025-02-13 at 4 55 10 PM](https://github.com/user-attachments/assets/52adec4f-7b55-4cb7-b845-baad7152a3e3)
It is possible to disable that too, but I figured it would be useful for development to still be able to see them.

## Type of change
[//]: # 'remove options that are not relevant'
- [x] CI enhancement ✨ 

## Checklist before requesting review
